VERSIONING.md

```diff
@@ -0,0 +1,145 @@
+itself in its Helm chart, so the major, minor, and patch version of any UXP
+release will always match the embedded Crossplane major, minor, and patch
+version. The following sections describe the UXP versioning strategy, including
+when branches are cut, dependency tags are incremented, and new versions are
+released.
+
+## Format
+
+The UXP version string is formatted as a semantic version with appended
+pre-release information:
+
+```
+v<major>.<minor>.<patch>-up.<components>
+```
+
+The `<major>`, `<minor>`, and `<patch>` segments each refer to the corresponding
+embedded Crossplane `<major>`, `<minor>`, and `<patch>` versions. The embedded
+Crossplane version may differ from upstream Crossplane (i.e. Upbound may release
+patch versions past the end of life date for a Crossplane branch), but if the
+specified version exists upstream, the contents will match exactly.
+
+The `<components>` segment is a numeric identifier describing the collective
+version of software components that surround Crossplane in a UXP release. The
+`-up.<components>` segment must be a [valid smever pre-release
+segment](https://semver.org/#spec-item-9). Pre-release information is used
+instead of [build metadata](https://semver.org/#spec-item-10) as pre-release
+precedence in honored in semantic versioning. The `<components>` numeric
+identifier only corresponds to the version of components for its preceding
+`<major>`, `<minor>`, `<patch>` versions. There is no relationship between
+`<components>` versions across `<major>`, `<minor>`, `<patch>` versions.
+
+An example version string for a UXP release that embeds Crossplane `v1.3.0`
+could be:
+
+```
+v1.3.0-up.2
+```
+
+Additional pre-release segments may be appended both before and after the
+`-up-<components>` segment. Any pre-release segments appended before will
+correspond to the embedded Crossplane version (e.g. `v1.4.0-rc.0-up.1`). Any
+pre-release segments appended after will correspond to a pre-release version of
+UXP itself (e.g. `v1.4.0-up.1-rc.0`). There should _never_ be both a Crossplane
+pre-release and UXP pre-release on the same version string (see [Guarantees
+section](#guarantees)).
+
+## Terms
+
+The following terms are used throughout this document and are defined as
+follows:
+
+- **UXP Version**: the entire semantic version for a UXP release (e.g.
+  `v1.2.2-up.3`).
+- **Crossplane Version**: the version of Crossplane embedded in the UXP release
+  (e.g. `v1.2.2`).
+- **Components Marker**: the version of the components surrounding upstream (or
+  temporarily forked) Crossplane (e.g. `-up.3`).
+
+## Guarantees
+
+![versioning-diagram](docs/media/versioning.png)
+
+The UXP versioning strategy can be expressed through a series of guarantees,
+each of which are elaborated in the subsequent sections. The diagram above
+illustrates examples of each guarantee in practice.
+
+1. The components marker on UXP main should always end in `-up.1` (i.e. no
+   components release candidates).
```

Comment:
Doesn't this counter the statement above?

> Additional pre-release segments may be appended both before and after the `-up-<components>` segment.

Reply:
Oh - I think I missed the key point that this is talking about the main branch specifically.